chaos/cluster.go

```diff
@@ -0,0 +1,264 @@
+package chaos
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+
+	"sync"
+
+	"github.com/coreos/ktestutil/utils"
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	// NodeRoleMasterLabel defines the master node's label.
+	NodeRoleMasterLabel = "node-role.kubernetes.io/master"
+	// NodeRoleWorkerLabel defines the worker node's label.
+	NodeRoleWorkerLabel = "node-role.kubernetes.io/node"
+
+	rebootCmd         = "sudo reboot"
+	systemUpCmd       = "sudo systemctl is-active docker"
+	kubeletDisableCmd = "sudo systemctl disable kubelet"
+	kubeletStartCmd   = "sudo systemctl enable kubelet && sudo systemctl start kubelet"
+)
+
+// Cluster is a simple abstraction that stores cluster nodes.
+type Cluster struct {
+	Masters []*v1.Node
+	Workers []*v1.Node
+	// Defaults to 0.
+	RebootAfter time.Duration
```

Comment:
to simulate a hard shutdown for x seconds, we probably should:

1. tear down network interface
2. kill kubelet
3. wait for x seconds
4. reboot

we need to tear down first since rebooting machine will send RST to all connections, and cause the other side to close tcp connections.